app/controllers/application_controller.rb

```diff
@@ -2,4 +2,62 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def my_order
+    unless cookies[:order].nil?
+      @order ||= Order.find(cookies.signed[:order])
+    else
+      @order = Order.new(status: "pending")
+    end
+  end
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def require_login
+    if current_user.nil?
+      flash[:error] = "You must be logged in to view this section"
+      redirect_to new_session_path
+    end
+  end
+
+  def not_require_login
```

Comment:
Method name was a little confusing when I encountered it. Perhaps something with more clarity, like logged_in or already_logged_in?